api/src/test/java/com/bjs/bjsapi/controllers/TopStudentsControllerTest.java

```diff
@@ -0,0 +1,236 @@
+package com.bjs.bjsapi.controllers;
+
+import com.bjs.bjsapi.database.model.Class;
+import com.bjs.bjsapi.database.model.Student;
+import com.bjs.bjsapi.database.model.helper.ClassBuilder;
+import com.bjs.bjsapi.database.model.helper.StudentBuilder;
+import com.bjs.bjsapi.database.repository.ClassRepository;
+import com.bjs.bjsapi.database.repository.StudentRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+@SpringJUnitConfig
+public class TopStudentsControllerTest {
+    @MockBean
+    private StudentRepository studentRepository;
+
+    @MockBean
+    private StudentRestController studentRestController;
+
+    @MockBean
+    private ClassRepository classRepository;
+
+    @MockBean
+    private TopStudentsController topStudentsController;
+
+    @BeforeEach
+    void setUp() {
+        topStudentsController = new TopStudentsController(studentRestController, studentRepository, classRepository);
+    }
+
+    @Test
```

Comment:
There may be a prettier way than making quite similar tests. I will look at this later more